// @bnb-chain/greenfield-cosmos-types/greenfield/virtualgroup/common.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.pickVGFStrategyToJSON = exports.pickVGFStrategyFromJSON = exports.PickVGFStrategyAmino = exports.PickVGFStrategySDKType = exports.PickVGFStrategy = exports.protobufPackage = void 0;
//@ts-nocheck
/* eslint-disable */
exports.protobufPackage = "greenfield.virtualgroup";
/** PickVGFStrategy represents the method for selecting the best global virtual group family based on the strategy */
var PickVGFStrategy;
(function (PickVGFStrategy) {
    PickVGFStrategy[PickVGFStrategy["Strategy_Maximize_Free_Store_Size"] = 0] = "Strategy_Maximize_Free_Store_Size";
    PickVGFStrategy[PickVGFStrategy["Strategy_Minimal_Free_Store_Size"] = 1] = "Strategy_Minimal_Free_Store_Size";
    PickVGFStrategy[PickVGFStrategy["Strategy_Oldest_Create_Time"] = 2] = "Strategy_Oldest_Create_Time";
    PickVGFStrategy[PickVGFStrategy["Strategy_Recentest_Create_Time"] = 3] = "Strategy_Recentest_Create_Time";
    PickVGFStrategy[PickVGFStrategy["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(PickVGFStrategy = exports.PickVGFStrategy || (exports.PickVGFStrategy = {}));
exports.PickVGFStrategySDKType = PickVGFStrategy;
exports.PickVGFStrategyAmino = PickVGFStrategy;
function pickVGFStrategyFromJSON(object) {
    switch (object) {
        case 0:
        case "Strategy_Maximize_Free_Store_Size":
            return PickVGFStrategy.Strategy_Maximize_Free_Store_Size;
        case 1:
        case "Strategy_Minimal_Free_Store_Size":
            return PickVGFStrategy.Strategy_Minimal_Free_Store_Size;
        case 2:
        case "Strategy_Oldest_Create_Time":
            return PickVGFStrategy.Strategy_Oldest_Create_Time;
        case 3:
        case "Strategy_Recentest_Create_Time":
            return PickVGFStrategy.Strategy_Recentest_Create_Time;
        case -1:
        case "UNRECOGNIZED":
        default:
            return PickVGFStrategy.UNRECOGNIZED;
    }
}
exports.pickVGFStrategyFromJSON = pickVGFStrategyFromJSON;
function pickVGFStrategyToJSON(object) {
    switch (object) {
        case PickVGFStrategy.Strategy_Maximize_Free_Store_Size:
            return "Strategy_Maximize_Free_Store_Size";
        case PickVGFStrategy.Strategy_Minimal_Free_Store_Size:
            return "Strategy_Minimal_Free_Store_Size";
        case PickVGFStrategy.Strategy_Oldest_Create_Time:
            return "Strategy_Oldest_Create_Time";
        case PickVGFStrategy.Strategy_Recentest_Create_Time:
            return "Strategy_Recentest_Create_Time";
        case PickVGFStrategy.UNRECOGNIZED:
        default:
            return "UNRECOGNIZED";
    }
}
exports.pickVGFStrategyToJSON = pickVGFStrategyToJSON;
