// @bnb-chain/greenfield-cosmos-types/greenfield/virtualgroup/tx.registry.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MessageComposer = exports.load = exports.registry = void 0;
const tx_1 = require("./tx");
exports.registry = [["/greenfield.virtualgroup.MsgCreateGlobalVirtualGroup", tx_1.MsgCreateGlobalVirtualGroup], ["/greenfield.virtualgroup.MsgDeleteGlobalVirtualGroup", tx_1.MsgDeleteGlobalVirtualGroup], ["/greenfield.virtualgroup.MsgDeposit", tx_1.MsgDeposit], ["/greenfield.virtualgroup.MsgWithdraw", tx_1.MsgWithdraw], ["/greenfield.virtualgroup.MsgSwapOut", tx_1.MsgSwapOut], ["/greenfield.virtualgroup.MsgSettle", tx_1.MsgSettle], ["/greenfield.virtualgroup.MsgUpdateParams", tx_1.MsgUpdateParams], ["/greenfield.virtualgroup.MsgStorageProviderExit", tx_1.MsgStorageProviderExit], ["/greenfield.virtualgroup.MsgCompleteStorageProviderExit", tx_1.MsgCompleteStorageProviderExit], ["/greenfield.virtualgroup.MsgCompleteSwapOut", tx_1.MsgCompleteSwapOut], ["/greenfield.virtualgroup.MsgCancelSwapOut", tx_1.MsgCancelSwapOut], ["/greenfield.virtualgroup.MsgReserveSwapIn", tx_1.MsgReserveSwapIn], ["/greenfield.virtualgroup.MsgCancelSwapIn", tx_1.MsgCancelSwapIn], ["/greenfield.virtualgroup.MsgCompleteSwapIn", tx_1.MsgCompleteSwapIn], ["/greenfield.virtualgroup.MsgStorageProviderForcedExit", tx_1.MsgStorageProviderForcedExit]];
const load = (protoRegistry) => {
    exports.registry.forEach(([typeUrl, mod]) => {
        protoRegistry.register(typeUrl, mod);
    });
};
exports.load = load;
exports.MessageComposer = {
    encoded: {
        createGlobalVirtualGroup(value) {
            return {
                typeUrl: "/greenfield.virtualgroup.MsgCreateGlobalVirtualGroup",
                value: tx_1.MsgCreateGlobalVirtualGroup.encode(value).finish()
            };
        },
        deleteGlobalVirtualGroup(value) {
            return {
                typeUrl: "/greenfield.virtualgroup.MsgDeleteGlobalVirtualGroup",
                value: tx_1.MsgDeleteGlobalVirtualGroup.encode(value).finish()
            };
        },
        deposit(value) {
            return {
                typeUrl: "/greenfield.virtualgroup.MsgDeposit",
                value: tx_1.MsgDeposit.encode(value).finish()
            };
        },
        withdraw(value) {
            return {
                typeUrl: "/greenfield.virtualgroup.MsgWithdraw",
                value: tx_1.MsgWithdraw.encode(value).finish()
            };
        },
        swapOut(value) {
            return {
                typeUrl: "/greenfield.virtualgroup.MsgSwapOut",
                value: tx_1.MsgSwapOut.encode(value).finish()
            };
        },
        settle(value) {
            return {
                typeUrl: "/greenfield.virtualgroup.MsgSettle",
                value: tx_1.MsgSettle.encode(value).finish()
            };
        },
        updateParams(value) {
            return {
                typeUrl: "/greenfield.virtualgroup.MsgUpdateParams",
                value: tx_1.MsgUpdateParams.encode(value).finish()
            };
        },
        storageProviderExit(value) {
            return {
                typeUrl: "/greenfield.virtualgroup.MsgStorageProviderExit",
                value: tx_1.MsgStorageProviderExit.encode(value).finish()
            };
        },
        completeStorageProviderExit(value) {
            return {
                typeUrl: "/greenfield.virtualgroup.MsgCompleteStorageProviderExit",
                value: tx_1.MsgCompleteStorageProviderExit.encode(value).finish()
            };
        },
        completeSwapOut(value) {
            return {
                typeUrl: "/greenfield.virtualgroup.MsgCompleteSwapOut",
                value: tx_1.MsgCompleteSwapOut.encode(value).finish()
            };
        },
        cancelSwapOut(value) {
            return {
                typeUrl: "/greenfield.virtualgroup.MsgCancelSwapOut",
                value: tx_1.MsgCancelSwapOut.encode(value).finish()
            };
        },
        reserveSwapIn(value) {
            return {
                typeUrl: "/greenfield.virtualgroup.MsgReserveSwapIn",
                value: tx_1.MsgReserveSwapIn.encode(value).finish()
            };
        },
        cancelSwapIn(value) {
            return {
                typeUrl: "/greenfield.virtualgroup.MsgCancelSwapIn",
                value: tx_1.MsgCancelSwapIn.encode(value).finish()
            };
        },
        completeSwapIn(value) {
            return {
                typeUrl: "/greenfield.virtualgroup.MsgCompleteSwapIn",
                value: tx_1.MsgCompleteSwapIn.encode(value).finish()
            };
        },
        storageProviderForcedExit(value) {
            return {
                typeUrl: "/greenfield.virtualgroup.MsgStorageProviderForcedExit",
                value: tx_1.MsgStorageProviderForcedExit.encode(value).finish()
            };
        }
    },
    withTypeUrl: {
        createGlobalVirtualGroup(value) {
            return {
                typeUrl: "/greenfield.virtualgroup.MsgCreateGlobalVirtualGroup",
                value
            };
        },
        deleteGlobalVirtualGroup(value) {
            return {
                typeUrl: "/greenfield.virtualgroup.MsgDeleteGlobalVirtualGroup",
                value
            };
        },
        deposit(value) {
            return {
                typeUrl: "/greenfield.virtualgroup.MsgDeposit",
                value
            };
        },
        withdraw(value) {
            return {
                typeUrl: "/greenfield.virtualgroup.MsgWithdraw",
                value
            };
        },
        swapOut(value) {
            return {
                typeUrl: "/greenfield.virtualgroup.MsgSwapOut",
                value
            };
        },
        settle(value) {
            return {
                typeUrl: "/greenfield.virtualgroup.MsgSettle",
                value
            };
        },
        updateParams(value) {
            return {
                typeUrl: "/greenfield.virtualgroup.MsgUpdateParams",
                value
            };
        },
        storageProviderExit(value) {
            return {
                typeUrl: "/greenfield.virtualgroup.MsgStorageProviderExit",
                value
            };
        },
        completeStorageProviderExit(value) {
            return {
                typeUrl: "/greenfield.virtualgroup.MsgCompleteStorageProviderExit",
                value
            };
        },
        completeSwapOut(value) {
            return {
                typeUrl: "/greenfield.virtualgroup.MsgCompleteSwapOut",
                value
            };
        },
        cancelSwapOut(value) {
            return {
                typeUrl: "/greenfield.virtualgroup.MsgCancelSwapOut",
                value
            };
        },
        reserveSwapIn(value) {
            return {
                typeUrl: "/greenfield.virtualgroup.MsgReserveSwapIn",
                value
            };
        },
        cancelSwapIn(value) {
            return {
                typeUrl: "/greenfield.virtualgroup.MsgCancelSwapIn",
                value
            };
        },
        completeSwapIn(value) {
            return {
                typeUrl: "/greenfield.virtualgroup.MsgCompleteSwapIn",
                value
            };
        },
        storageProviderForcedExit(value) {
            return {
                typeUrl: "/greenfield.virtualgroup.MsgStorageProviderForcedExit",
                value
            };
        }
    },
    toJSON: {
        createGlobalVirtualGroup(value) {
            return {
                typeUrl: "/greenfield.virtualgroup.MsgCreateGlobalVirtualGroup",
                value: tx_1.MsgCreateGlobalVirtualGroup.toJSON(value)
            };
        },
        deleteGlobalVirtualGroup(value) {
            return {
                typeUrl: "/greenfield.virtualgroup.MsgDeleteGlobalVirtualGroup",
                value: tx_1.MsgDeleteGlobalVirtualGroup.toJSON(value)
            };
        },
        deposit(value) {
            return {
                typeUrl: "/greenfield.virtualgroup.MsgDeposit",
                value: tx_1.MsgDeposit.toJSON(value)
            };
        },
        withdraw(value) {
            return {
                typeUrl: "/greenfield.virtualgroup.MsgWithdraw",
                value: tx_1.MsgWithdraw.toJSON(value)
            };
        },
        swapOut(value) {
            return {
                typeUrl: "/greenfield.virtualgroup.MsgSwapOut",
                value: tx_1.MsgSwapOut.toJSON(value)
            };
        },
        settle(value) {
            return {
                typeUrl: "/greenfield.virtualgroup.MsgSettle",
                value: tx_1.MsgSettle.toJSON(value)
            };
        },
        updateParams(value) {
            return {
                typeUrl: "/greenfield.virtualgroup.MsgUpdateParams",
                value: tx_1.MsgUpdateParams.toJSON(value)
            };
        },
        storageProviderExit(value) {
            return {
                typeUrl: "/greenfield.virtualgroup.MsgStorageProviderExit",
                value: tx_1.MsgStorageProviderExit.toJSON(value)
            };
        },
        completeStorageProviderExit(value) {
            return {
                typeUrl: "/greenfield.virtualgroup.MsgCompleteStorageProviderExit",
                value: tx_1.MsgCompleteStorageProviderExit.toJSON(value)
            };
        },
        completeSwapOut(value) {
            return {
                typeUrl: "/greenfield.virtualgroup.MsgCompleteSwapOut",
                value: tx_1.MsgCompleteSwapOut.toJSON(value)
            };
        },
        cancelSwapOut(value) {
            return {
                typeUrl: "/greenfield.virtualgroup.MsgCancelSwapOut",
                value: tx_1.MsgCancelSwapOut.toJSON(value)
            };
        },
        reserveSwapIn(value) {
            return {
                typeUrl: "/greenfield.virtualgroup.MsgReserveSwapIn",
                value: tx_1.MsgReserveSwapIn.toJSON(value)
            };
        },
        cancelSwapIn(value) {
            return {
                typeUrl: "/greenfield.virtualgroup.MsgCancelSwapIn",
                value: tx_1.MsgCancelSwapIn.toJSON(value)
            };
        },
        completeSwapIn(value) {
            return {
                typeUrl: "/greenfield.virtualgroup.MsgCompleteSwapIn",
                value: tx_1.MsgCompleteSwapIn.toJSON(value)
            };
        },
        storageProviderForcedExit(value) {
            return {
                typeUrl: "/greenfield.virtualgroup.MsgStorageProviderForcedExit",
                value: tx_1.MsgStorageProviderForcedExit.toJSON(value)
            };
        }
    },
    fromJSON: {
        createGlobalVirtualGroup(value) {
            return {
                typeUrl: "/greenfield.virtualgroup.MsgCreateGlobalVirtualGroup",
                value: tx_1.MsgCreateGlobalVirtualGroup.fromJSON(value)
            };
        },
        deleteGlobalVirtualGroup(value) {
            return {
                typeUrl: "/greenfield.virtualgroup.MsgDeleteGlobalVirtualGroup",
                value: tx_1.MsgDeleteGlobalVirtualGroup.fromJSON(value)
            };
        },
        deposit(value) {
            return {
                typeUrl: "/greenfield.virtualgroup.MsgDeposit",
                value: tx_1.MsgDeposit.fromJSON(value)
            };
        },
        withdraw(value) {
            return {
                typeUrl: "/greenfield.virtualgroup.MsgWithdraw",
                value: tx_1.MsgWithdraw.fromJSON(value)
            };
        },
        swapOut(value) {
            return {
                typeUrl: "/greenfield.virtualgroup.MsgSwapOut",
                value: tx_1.MsgSwapOut.fromJSON(value)
            };
        },
        settle(value) {
            return {
                typeUrl: "/greenfield.virtualgroup.MsgSettle",
                value: tx_1.MsgSettle.fromJSON(value)
            };
        },
        updateParams(value) {
            return {
                typeUrl: "/greenfield.virtualgroup.MsgUpdateParams",
                value: tx_1.MsgUpdateParams.fromJSON(value)
            };
        },
        storageProviderExit(value) {
            return {
                typeUrl: "/greenfield.virtualgroup.MsgStorageProviderExit",
                value: tx_1.MsgStorageProviderExit.fromJSON(value)
            };
        },
        completeStorageProviderExit(value) {
            return {
                typeUrl: "/greenfield.virtualgroup.MsgCompleteStorageProviderExit",
                value: tx_1.MsgCompleteStorageProviderExit.fromJSON(value)
            };
        },
        completeSwapOut(value) {
            return {
                typeUrl: "/greenfield.virtualgroup.MsgCompleteSwapOut",
                value: tx_1.MsgCompleteSwapOut.fromJSON(value)
            };
        },
        cancelSwapOut(value) {
            return {
                typeUrl: "/greenfield.virtualgroup.MsgCancelSwapOut",
                value: tx_1.MsgCancelSwapOut.fromJSON(value)
            };
        },
        reserveSwapIn(value) {
            return {
                typeUrl: "/greenfield.virtualgroup.MsgReserveSwapIn",
                value: tx_1.MsgReserveSwapIn.fromJSON(value)
            };
        },
        cancelSwapIn(value) {
            return {
                typeUrl: "/greenfield.virtualgroup.MsgCancelSwapIn",
                value: tx_1.MsgCancelSwapIn.fromJSON(value)
            };
        },
        completeSwapIn(value) {
            return {
                typeUrl: "/greenfield.virtualgroup.MsgCompleteSwapIn",
                value: tx_1.MsgCompleteSwapIn.fromJSON(value)
            };
        },
        storageProviderForcedExit(value) {
            return {
                typeUrl: "/greenfield.virtualgroup.MsgStorageProviderForcedExit",
                value: tx_1.MsgStorageProviderForcedExit.fromJSON(value)
            };
        }
    },
    fromPartial: {
        createGlobalVirtualGroup(value) {
            return {
                typeUrl: "/greenfield.virtualgroup.MsgCreateGlobalVirtualGroup",
                value: tx_1.MsgCreateGlobalVirtualGroup.fromPartial(value)
            };
        },
        deleteGlobalVirtualGroup(value) {
            return {
                typeUrl: "/greenfield.virtualgroup.MsgDeleteGlobalVirtualGroup",
                value: tx_1.MsgDeleteGlobalVirtualGroup.fromPartial(value)
            };
        },
        deposit(value) {
            return {
                typeUrl: "/greenfield.virtualgroup.MsgDeposit",
                value: tx_1.MsgDeposit.fromPartial(value)
            };
        },
        withdraw(value) {
            return {
                typeUrl: "/greenfield.virtualgroup.MsgWithdraw",
                value: tx_1.MsgWithdraw.fromPartial(value)
            };
        },
        swapOut(value) {
            return {
                typeUrl: "/greenfield.virtualgroup.MsgSwapOut",
                value: tx_1.MsgSwapOut.fromPartial(value)
            };
        },
        settle(value) {
            return {
                typeUrl: "/greenfield.virtualgroup.MsgSettle",
                value: tx_1.MsgSettle.fromPartial(value)
            };
        },
        updateParams(value) {
            return {
                typeUrl: "/greenfield.virtualgroup.MsgUpdateParams",
                value: tx_1.MsgUpdateParams.fromPartial(value)
            };
        },
        storageProviderExit(value) {
            return {
                typeUrl: "/greenfield.virtualgroup.MsgStorageProviderExit",
                value: tx_1.MsgStorageProviderExit.fromPartial(value)
            };
        },
        completeStorageProviderExit(value) {
            return {
                typeUrl: "/greenfield.virtualgroup.MsgCompleteStorageProviderExit",
                value: tx_1.MsgCompleteStorageProviderExit.fromPartial(value)
            };
        },
        completeSwapOut(value) {
            return {
                typeUrl: "/greenfield.virtualgroup.MsgCompleteSwapOut",
                value: tx_1.MsgCompleteSwapOut.fromPartial(value)
            };
        },
        cancelSwapOut(value) {
            return {
                typeUrl: "/greenfield.virtualgroup.MsgCancelSwapOut",
                value: tx_1.MsgCancelSwapOut.fromPartial(value)
            };
        },
        reserveSwapIn(value) {
            return {
                typeUrl: "/greenfield.virtualgroup.MsgReserveSwapIn",
                value: tx_1.MsgReserveSwapIn.fromPartial(value)
            };
        },
        cancelSwapIn(value) {
            return {
                typeUrl: "/greenfield.virtualgroup.MsgCancelSwapIn",
                value: tx_1.MsgCancelSwapIn.fromPartial(value)
            };
        },
        completeSwapIn(value) {
            return {
                typeUrl: "/greenfield.virtualgroup.MsgCompleteSwapIn",
                value: tx_1.MsgCompleteSwapIn.fromPartial(value)
            };
        },
        storageProviderForcedExit(value) {
            return {
                typeUrl: "/greenfield.virtualgroup.MsgStorageProviderForcedExit",
                value: tx_1.MsgStorageProviderForcedExit.fromPartial(value)
            };
        }
    }
};
